python/lsst/ap/association/packageAlerts.py

```diff
@@ -77,6 +93,52 @@ def __init__(self, **kwargs):
         self.alertSchema = alertPack.Schema.from_uri(self.config.schemaFile)
         os.makedirs(self.config.alertWriteLocation, exist_ok=True)
 
+        if self.config.doProduceAlerts:
+
+            if confluent_kafka is not None:
+
+                self.password = os.getenv("AP_KAFKA_PRODUCER_PASSWORD")
+                self.username = os.getenv("AP_KAFKA_PRODUCER_USERNAME")
+                self.server = os.getenv("AP_KAFKA_SERVER")
+                self.kafkaTopic = os.getenv("AP_KAFKA_TOPIC")
+
+                if not self.password:
+                    raise ValueError("Kafka password environment variable was not set.")
+                if not self.username:
+                    raise ValueError("Kafka username environment variable was not set.")
+                if not self.server:
+                    raise ValueError("Kafka server environment variable was not set.")
+                if not self.kafkaTopic:
+                    raise ValueError("Kafka topic environment variable was not set.")
+
+                # confluent_kafka configures all of its classes with dictionaries. This one
+                # sets up the bare minimum that is needed.
+                self.kafkaConfig = {
+                    # This is the URL to use to connect to the Kafka cluster.
+                    "bootstrap.servers": self.server,
+                    # These next two properties tell the Kafka client about the specific
+                    # authentication and authorization protocols that should be used when
+                    # connecting.
+                    "security.protocol": "SASL_PLAINTEXT",
+                    "sasl.mechanisms": "SCRAM-SHA-512",
+                    # The sasl.username and sasl.password are passed through over
+                    # SCRAM-SHA-512 auth to connect to the cluster. The username is not
+                    # sensitive, but the password is (of course) a secret value which
+                    # should never be committed to source code.
+                    "sasl.username": self.username,
+                    "sasl.password": self.password,
+                    # Batch size limits the largest size of a kafka alert that can be sent.
+                    # We set the batch size to 2 Mb.
+                    "batch.size": 2097152,
+                    "linger.ms": 5,
+                }
+
+                self.producer = confluent_kafka.Producer(**self.kafkaConfig)
+
+            else:
+                self.log.error("Produce alerts is set but confluent_kafka is not present in "
+                               "the environment. Alerts will not be sent to the alert stream.")
```

Comment:
Shouldn't this be a `raise` (probably RuntimeError for now?)? If someone sets that to True, they want to produce alerts.

Reply:
I assume then it would be fine to remove the `self.log.error` since `raise` would be logged?

Reply:
Yes, if we just raise we don't need to log.